azext_iot/dps/providers/device_registration.py

```diff
@@ -0,0 +1,258 @@
+                pass_phrase=passphrase or ""
+            )
+        elif certificate_file or key_file:
+            raise RequiredArgumentMissingError(CERTIFICATE_FILE_ERROR)
+        # Retrieve the attestation if nothing is provided.
+        else:
+            self._get_attestation_params(enrollment_group_id=enrollment_group_id)
+
+    def _get_idscope(self) -> str:
+        discovery = DPSDiscovery(self.cmd)
+        target = discovery.get_target(
+            self.dps_name,
+            self.resource_group_name,
+            login=self.login,
+            auth_type=self.auth_type_dataplane,
+        )
+        if target.get("idscope"):
+            return target["idscope"]
+        # If cstring is used, will need to retrieve the id scope manually
+        dps_name = target['entity'].split(".")[0]
+        return discovery.get_id_scope(resource_name=dps_name)
+
+    def _get_dps_device_sdk(
+        self
+    ):
+        from azure.iot.device import ProvisioningDeviceClient
+        if self.device_symmetric_key:
+            return ProvisioningDeviceClient.create_from_symmetric_key(
+                provisioning_host=IOTDPS_PROVISIONING_HOST,
+                registration_id=self.registration_id,
+                id_scope=self.id_scope,
+                symmetric_key=self.device_symmetric_key,
+            )
+        elif self.certificate:
+            try:
+                return ProvisioningDeviceClient.create_from_x509_certificate(
+                    provisioning_host=IOTDPS_PROVISIONING_HOST,
+                    registration_id=self.registration_id,
+                    id_scope=self.id_scope,
+                    x509=self.certificate,
+                )
+            except Exception as e:
+                if hasattr(e, "reason"):
+                    raise InvalidArgumentValueError(f"Could not open certificate files: {e.reason}.")
+
+    def create(
+        self,
+        payload: str = None,
+    ):
+        try:
+            self.sdk.provisioning_payload = payload
+            registration_result = self.sdk.register()
+            return {
+                "operationId": registration_result.operation_id,
+                "status": registration_result.status,
+                "registrationState": {
+                    "registrationId": self.registration_id,
+                    "deviceId": registration_result.registration_state.device_id,
+                    "assignedHub": registration_result.registration_state.assigned_hub,
+                    "substatus": registration_result.registration_state.sub_status,
+                    "createdDateTimeUtc": registration_result.registration_state.created_date_time,
+                    "lastUpdatedDateTimeUtc": registration_result.registration_state.last_update_date_time,
+                    "etag": registration_result.registration_state.etag,
+                    "responsePayload": registration_result.registration_state.response_payload,
+                }
```

Comment:
so this is the result of using the device sdk rather than a generated sdk. I included registration id cause it is a value that I saw in the generated sdk but the device sdk omits this. Furthermore, the device sdk returns an object whose string representation is just some values seperated by \n like so:

enabled
device_id
hub_name
Provisioned to Initial Assignment


Hence the recreation of the object